### PR TITLE
Re-enable build of matrix completion and betweenness centrality

### PR DESCRIPTION
**Pull request by ddn0**

### CODE REVIEW

File: lonestar/analytics/distributed/betweennesscentrality/bc_level.cpp [hunk 1]
```diff
@@ -220,48 +225,48 @@ struct InitializeIteration {
...
+      Graph* graph, galois::DGAccumulator<uint32_t>& dga, uint32_t roundNum)
+      : graph_(graph), dga_(dga), local_r_(roundNum) {}
 
   /**
    * Level by level BFS while also finding number of shortest paths to a
    * particular node in the BFS tree.
    *
-   * @param _graph Graph to use
+   * @param graph Graph to use
    * @param _dga distributed accumulator
```
Comment (gurbinder533) [inline lonestar/analytics/distributed/betweennesscentrality/bc_level.cpp:241]: _dga --> dga

Comment (ddn0) [reply lonestar/analytics/distributed/betweennesscentrality/bc_level.cpp]: Fixed; I did a quick grep for "param _" and there doesn't appear to be any other instances in the dist analytics applications.

File: lonestar/analytics/distributed/betweennesscentrality/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#app_dist(bc_level betweennesscentrality-level)
+app_dist(bc_level betweennesscentrality-level)
```
Comment (roshandathathri) [inline lonestar/analytics/distributed/betweennesscentrality/CMakeLists.txt:1]: Ctests don't work for this? Don't see GALOIS_LOG_FATAL in the code.

File: libgalois/include/galois/DynamicBitset.h [hunk 3]
```diff
@@ -44,9 +44,10 @@ namespace galois {
 class GALOIS_EXPORT DynamicBitSet {
   galois::PODResizeableArray<galois::CopyableAtomic<uint64_t>> bitvec;
   size_t num_bits{0};
-  static constexpr uint32_t bits_uint64 = sizeof(uint64_t) * CHAR_BIT;
 
 public:
+  static constexpr uint32_t bits_uint64 = sizeof(uint64_t) * CHAR_BIT;
```
Comment (roshandathathri) [inline libgalois/include/galois/DynamicBitset.h:49]: Why is this required?

**Review by roshandathathri — APPROVED** [review]
LGTM

Did you use clang-tidy for this? Or did you do this manually?

**Review by gurbinder533 — APPROVED** [review]
LGTM.

File: lonestar/analytics/distributed/betweennesscentrality/CMakeLists.txt (hunk 2 above)
Comment (ddn0) [reply lonestar/analytics/distributed/betweennesscentrality/CMakeLists.txt:1]: add_test_dist is commented out below, so cmake will compile but no tests will run:

```
#add_test_dist(betweennesscentrality-level-dist NO_ASYNC INPUT rmat15 INPUT_URI ${BASEINPUT}/scalefree/rmat15.gr -graphTranspose=${BASEINPUT}/scalefree/transpose/rmat15.tgr -numOfSources=4)
```

Though for bc-level, I think we could re-enable the tests because it doesn't require non-trivial data. Let me take a look.

File: libgalois/include/galois/DynamicBitset.h (hunk 3 above)
Comment (ddn0) [reply libgalois/include/galois/DynamicBitset.h:49]: bc_mr has its own bitset implementation that derives from DynamicBitSet and it needs access to this field. In general, I'd prefer extension by composition rather than inheritance because DynamicBitSet and bc_mr's subclass aren't really designed for clean extension by others. In this particular case, DynamicBitSet already has a public `get_vec` so it seems reasonable to also make `bits_uint64` public; otherwise, it is not possible for someone to interpret the return value of `get_vec`.

If DynamicBitSet kept `bitvec` private (which I think is a better interface), I would have just cloned/copied DynamicBitSet into the bc_mr application.

**Review by ddn0 — COMMENTED** [review]
> Did you use clang-tidy for this? Or did you do this manually?

I don't think clang-tidy can automate this kind of change (except for maybe automating the renames one-by-one). I did this manually.

File: lonestar/analytics/distributed/betweennesscentrality/CMakeLists.txt (hunk 2 above)
Comment (ddn0) [reply lonestar/analytics/distributed/betweennesscentrality/CMakeLists.txt:1]: Yup; the test bc_level seems to work okay.

File: libgalois/include/galois/DynamicBitset.h (hunk 3 above)
Comment (roshandathathri) [reply libgalois/include/galois/DynamicBitset.h:49]: Ok, makes sense.

File: lonestar/analytics/distributed/betweennesscentrality/CMakeLists.txt (hunk 2 above)
Comment (roshandathathri) [reply lonestar/analytics/distributed/betweennesscentrality/CMakeLists.txt:1]: Cool
